buildSrc/src/main/groovy/com/jakewharton/kotlingen/KotlinGenTask.groovy

```diff
@@ -0,0 +1,284 @@
+package com.jakewharton.kotlingen
+
+import com.github.javaparser.JavaParser
+import com.github.javaparser.ast.CompilationUnit
+import com.github.javaparser.ast.ImportDeclaration
+import com.github.javaparser.ast.PackageDeclaration
+import com.github.javaparser.ast.TypeParameter
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration
+import com.github.javaparser.ast.body.MethodDeclaration
+import com.github.javaparser.ast.body.ModifierSet
+import com.github.javaparser.ast.body.Parameter
+import com.github.javaparser.ast.type.*
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter
+import com.google.common.collect.ImmutableList
+import org.gradle.api.tasks.SourceTask
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.incremental.IncrementalTaskInputs
+import sun.reflect.generics.reflectiveObjects.NotImplementedException
+
+import java.nio.file.Files
+import java.nio.file.Path
+
+/**
+ * Task that generates Kotlin bindings for the Rx*.java implementations
+ *
+ * To debug this, you can use the following command and attach a remote debugger to port 5005
+ * ./gradlew clean build --no-daemon -Dorg.gradle.debug=true
+ */
+class KotlinGenTask extends SourceTask {
+
+  /** Regex used for finding references in javadoc links */
+  private static final String DOC_LINK_REGEX = "[0-9A-Za-z._]*"
+
+  /**
+   * These are imports of classes that Kotlin advises against using and are replaced in
+   * {@link #resolveKotlinTypeByName}
+   */
+  private static final List<String> IGNORED_IMPORTS = ImmutableList.of(
+          "java.util.List"
+  )
+
+  @TaskAction
+  def generate(IncrementalTaskInputs inputs) {
+    getSource().each { generateKotlin(it) }
+  }
+
+  void generateKotlin(File file) {
+    String outputPath = file.parent.replace("java", "kotlin").replace("/src", "-kotlin/src")
```

Comment:
Since the kotlin library contains 100% generated code, we should clear the output dir before writing to it. This will prevent old classes from hanging around on rename or delete.